Clean up Dutch local flavor.

jieter opened the pull request:
This is my initial attempt on #150. As of now, nothing is (or at least should be) renamed or deprecated. This is what I did:

 - Move validators to validators.py
 - Add model fields for existing form fields
 - Add form fields for existing model fields
 - Move tests to `tests/test_nl/`
 - Increase test coverage.
 - Tried to make the error messages a bit more consistent.

## Known issues / TODOs:
 - Add docstrings
 - Tests fail with django 1.5: `RegExvalidator` somehow fails with message `Enter a valid value.` and not `Enter a valid zip code.`. Any pointers to fix this?

Please review!

jieter commented:
@erikr @benkonrath ?

benkonrath commented:
@jieter Sorry, this fell off my radar. I'll try to look at it later today.

benkonrath commented on localflavor/nl/validators.py:
```diff
@@ -0,0 +1,127 @@
+    """
+    Validation for Dutch Sofinummers.
+
+    TODO: consider renaming to BSN
```
Yeah, I agree that it should be changed to BSN but let's do this as separate issue.

benkonrath commented on localflavor/nl/models.py:
```diff
@@ -1,62 +1,88 @@
+    """
+    Dutch phone number model field
+    """
+    description = _('Dutch phone number')
```
The comment is redundant with the description right there.

jieter replied:
This is in the code multiple times, should I remove them all? Are they not useful while generating the docs?

benkonrath replied:
Yes, good point, they are used for the docs. Please ignore my comment.

benkonrath commented on localflavor/nl/models.py:
```diff
@@ -1,62 +1,88 @@
+
+
+class NLProvinceField(models.CharField):
+    description = _('Dutch social security number (SoFi)')
```
This is the NLProvinceField, not the SoFi field. It's probably just a cut 'n paste error. :)

benkonrath commented:
Besides those small points, it looks good. If you can fix these points, figure why the tests fail on 1.5 and get the docstrings written, then we can try to get it in the next release which is due shortly. Otherwise we'll shoot for the release after next when 1.5 support is dropped as per #170.

benkonrath commented on localflavor/nl/models.py:
```diff
@@ -1,62 +1,88 @@
-                                                                message=self.default_error_messages['invalid'])
-        self.no_leading_zeros_regex = re.compile('[1-9]+')
+    def to_python(self, value):
+        return '%s %s' % (value[:4], value[4:])
```
I found a problem here. This is adding an extra space when the model field is used with the form field. For example, if I enter '1000AA', I will get the error message: 'Ensure this value has at most 7 characters (it has 8).' because the form adds a space and then this adds a space. You'll need to check if the space is already there before adding it. You should also add a test for this. Thanks.

jieter replied:
Shouldn't we just save it in the database without space and present it with a space?

jieter replied:
Hmm, this seems to be fixed by the last commit I pushed.

jieter commented:
Thanks for the feedback, I'll come back to this, hopefully next week. The delay pushed it somewhat down on my todo-list...

benkonrath commented:
Understandable. I'll keep my eyes peeled for your update. Thanks.

jieter commented:
Just pushed some improvements. I'm not going to invest time in adding 1.5 support if it's about to be removed anyway, so I'll guess we have to wait until after the next release then?

benkonrath commented:
I think it makes sense to wait until localflavor 1.2 is released so that we can skip trying to fix the errors with Django 1.5. Although I'm not sure when that's going to happen.

The bug in the NL postal code model field in fact fixed but it would be nice to rework that bit of code to be more like the IBAN model. 

You should make sure you get 100% test coverage. There's a bit of the NL postal code model field that isn't tested and the bank account model field isn't tested as well. I know that we want to  deprecate and drop the EU bank account fields but we should keep the tests in until the fields are removed.

Thanks for your work on this.

jieter commented:
- Just noticed that I added forms.NLBankaccountField in this branch, so I removed it. We should not add new code to deprecate it immediately.
- re-added the test for the bank account model field, you're right about that.
- test coverage is now at 100%.
- the Zip code form field needs the six.string_types check because apparently an array might be passed.

benkonrath commented:
Nice catch on the forms.NLBankaccountField - I did see it but I just assumed it was already there. Nice work here - l'll merge this after localflavor 1.2 is released.

benkonrath commented:
I forgot to mention that you should rebase this and squash all your commits into one. Thanks.

jieter commented:
yes, will do. And what about the `.. versionadded:: TODO` markers?

benkonrath commented:
We'll have to wait to see what the version will be after 1.2. You can wait to squash everything once the `versionadded` comments are in.

jieter commented:
Will squash again then.

jieter commented:
All tests pass now, should I extend the docstrings with `..versionadded::1.3` now?

benkonrath commented:
Yeah, `..versionadded:: 1.3` would be great for any new classes.

Can you also add an entry to the changelog for this PR? The moved validator classes should be specifically mentioned as a breaking change so users know they will need to adjust their imports. I want to setup a deprecation policy for localflavor but we can sort that out before the next release.

The final (small) change it to reset the commit date. Simply running `git commit --amend --reset-author` on your branch should take care of it.

jieter commented:
This should do it.

benkonrath commented:
@jieter Thanks for your work on this!

jieter commented:
@benkonrath any idea on when a version containing this PR is going to be released to pypi?

benkonrath commented:
@jieter Within the next week or two I hope. See my comment about this in #205.

jieter commented:
I see, thanks for the update
